server/auth/range_perm_cache.go

```diff
@@ -155,3 +157,51 @@ type unifiedRangePermissions struct {
 	readPerms  adt.IntervalTree
 	writePerms adt.IntervalTree
 }
+
+// Constraints related to key range
```

Comment:
Does same rules apply for making a range request?

Reply:
Yes, the same rules are used by range, watch and delete.

Reply:
Could they use the same function to validate the range?

Reply:
I'm more worried about consistency, feel free to create an issue with the suggestion.